plugins/stopwatch.rst

```diff
@@ -1,4 +1,27 @@
 Stopwatch Plugin
 ================
 
-TODO: explain the stopwatch plugin
+The ``StopwatchPlugin`` records the duration of HTTP requests with a
+``Symfony\Component\Stopwatch\Stopwatch`` instance::
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Client\Plugin\PluginClient;
+    use Http\Client\Plugin\StopwatchPlugin;
+    use Symfony\Component\Stopwatch\Stopwatch;
+
+    $stopwatch = new Stopwatch();
+
+    $stopwatchPlugin = new StopwatchPlugin($stopwatch);
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$stopwatchPlugin]
+    );
+
+    // ...
+
+    foreach ($stopwatch->getSections() as $section) {
+        foreach ($section->getEvents() as $name => $event) {
+            echo sprintf('Request %s took %s ms and used %s bytes of memory', $name, $event->getDuration(), $event->getMemory());
+        }
+    }
```

Comment:
Don't know if this should be here, but Stopwatch plugin can be inconsistent when using async (not the real time, neither memory as it can change if user is doint another async operation). Maybe adding a warning about that (not sure also, don't want to complicate too much the documentation)